neuralmonkey/model/sequence.py

```diff
@@ -0,0 +1,192 @@
+        # the data_id string does not need to be the same across experiments
+        return [
+            tf.get_variable(
+                name="embedding_matrix_{}".format(i),
+                shape=[vocab_size, emb_size],
+                initializer=tf.random_normal_initializer(stddev=0.01))
+            for i, (data_id, vocab_size, emb_size) in enumerate(zip(
+                self.data_ids, self.vocabulary_sizes, self.embedding_sizes))]
+
+    @tensor
+    def data(self) -> tf.Tensor:
+        embedded_factors = [
+            tf.nn.embedding_lookup(embedding_matrix, factor)
+            for factor, embedding_matrix in zip(
+                self.input_factors, self.embedding_matrices)]
+
+        return tf.concat(embedded_factors, 2)
+
+    @property
+    def dimension(self) -> int:
+        return sum(self.embedding_sizes)
+
+    def feed_dict(self, dataset: Dataset, train: bool = False) -> FeedDict:
+        fd = {}  # type: FeedDict
+
+        # for checking the lengths of individual factors
+        arr_strings = []
+        last_paddings = None
+
+        for factor_plc, name, vocabulary in zip(
+                self.input_factors, self.data_ids, self.vocabularies):
+            factors = dataset.get_series(name)
+            vectors, paddings = vocabulary.sentences_to_tensor(
+                list(factors), self.max_length, pad_to_max_len=False,
+                train_mode=train)
+
+            fd[factor_plc] = list(zip(*vectors))
+
+            arr_strings.append(paddings.tostring())
+            last_paddings = paddings
+
+        if len(set(arr_strings)) > 1:
+            raise ValueError("The lenghts of factors do not match")
+
+        fd[self.mask] = list(zip(*last_paddings))
+
+        return fd
+
+
+class EmbeddedSequence(EmbeddedFactorSequence):
+    def __init__(self,
+                 name: str,
+                 vocabulary: Vocabulary,
+                 data_id: str,
+                 embedding_size: int,
+                 max_length: int = None,
+                 save_checkpoint: str = None,
+                 load_checkpoint: str = None) -> None:
+        EmbeddedFactorSequence.__init__(
+            self,
+            name=name,
+            vocabularies=[vocabulary],
+            data_ids=[data_id],
+            embedding_sizes=[embedding_size],
+            max_length=max_length,
+            save_checkpoint=save_checkpoint,
+            load_checkpoint=load_checkpoint)
+
+    # pylint: disable=unsubscriptable-object
+    @property
+    def inputs(self) -> tf.Tensor:
+        return self.input_factors[0]
+
+    @property
+    def embedding_matrix(self) -> tf.Tensor:
+        return self.embedding_matrices[0]
+    # pylint: enable=unsubscriptable-object
+
+    @property
+    def vocabulary(self) -> Vocabulary:
+        return self.vocabularies[0]
+
+    @property
+    def data_id(self) -> str:
+        return self.data_ids[0]
+
+    def feed_dict(self, dataset: Dataset, train: bool = False) -> FeedDict:
```

Comment:
Tohle se nemusí přece vůbec overridovat, ta metoda by měla fungovat out of the box i z rodiče, ne?